src/effects/native/phasereffect.cpp

```diff
@@ -0,0 +1,199 @@
+    Q_UNUSED(enableState);
+    Q_UNUSED(groupFeatures);
+    Q_UNUSED(sampleRate);
+
+    CSAMPLE frequency = m_pLFOFrequencyParameter->value();
+    CSAMPLE depth = m_pDepthParameter->value();
+    CSAMPLE feedback = m_pFeedbackParameter->value();
+    CSAMPLE range = m_pRangeParameter->value();
+    int stages = m_pStagesParameter->value();
+
+    CSAMPLE* oldInLeft = pState->oldInLeft;
+    CSAMPLE* oldOutLeft = pState->oldOutLeft;
+    CSAMPLE* oldInRight = pState->oldInRight;
+    CSAMPLE* oldOutRight = pState->oldOutRight;
+
+    //Using two sets of coefficients for left and right channel 
+    CSAMPLE filterCoefLeft = 0;
+    CSAMPLE filterCoefRight = 0;
+
+    //If changing the 'Rate' knob, the phaser will use the initial parameters    
+    //until a desired frequency is chosen (to avoid bubbling noises)
+    if (frequency != pState->frequency) {
+        pState->wait = 0;
+        pState->frequency = frequency; 
+        pState->change = true;
+    }
+
+    if (pState->change) {
+        pState->wait++;
+        if (pState->wait == pState->limit) {
+            pState->change = false;
+            pState->oldFrequency = frequency;
+        } else {
+            frequency = pState->oldFrequency;
+        }
+    }
+
+    CSAMPLE left = 0, right = 0;
+    CSAMPLE leftPhase, rightPhase;
+    CSAMPLE freqSkip = frequency * 2.0 * M_PI / sampleRate;
+
+    int stereoCheck = m_pStereoParameter->value();
+    int counter = 0;
+
+    const int kChannels = 2;
+    for (unsigned int i = 0; i < numSamples; i += kChannels) {
+        pState->time++;
+        left = pInput[i] + tanh(left * feedback); 
+        right = pInput[i + 1] + tanh(right * feedback);
+
+        //For stereo enabled, the channels are out of phase
+        leftPhase = fmodf(freqSkip * pState->time, 2.0 * M_PI);
+        rightPhase = fmodf(freqSkip * pState->time + M_PI * stereoCheck, 2.0 * M_PI);
+
+        //Updating filter coefficients once every 'updateCoef' samples to avoid
+        //extra computing
+        if ((counter++) % updateCoef == 0) {
+                CSAMPLE delayLeft = 0.5 + 0.5 * sin(leftPhase);
+                CSAMPLE delayRight = 0.5 + 0.5 * sin(rightPhase);
+                
+                //Coefficient computing based on the following:
+                //https://ccrma.stanford.edu/~jos/pasp/Classic_Virtual_Analog_Phase.html
+                CSAMPLE wLeft = range * delayLeft;
+                CSAMPLE wRight = range * delayRight;
+
+                CSAMPLE tanwLeft = tanh(wLeft / 2);
+                CSAMPLE tanwRight = tanh(wRight / 2);
+
+                filterCoefLeft = (1.0 - tanwLeft) / (1.0 + tanwLeft);
+                filterCoefRight = (1.0 - tanwRight) / (1.0 + tanwRight);
+        }
+       
+        left = processSample(left, oldInLeft, oldOutLeft, filterCoefLeft, stages);
+        right = processSample(right, oldInRight, oldOutRight, filterCoefRight, stages);
+
+        //Computing output combining the original and processed sample
```

Comment:
Nit: We use a space after // to distinguish comments from deactivated code. Please add a space here and in similar places.